observation_portal/userrequests/conversion.py

```diff
@@ -0,0 +1,249 @@
+import os
+import copy
+import requests
+
+VALHALLA_API_TOKEN = os.getenv('VALHALLA_API_TOKEN', '')
+VALHALLA_URL = os.getenv('VALHALLA_URL', 'http://valhalladev.lco.gtn/api/userrequests/')
+if not VALHALLA_URL.endswith('/'):
+    VALHALLA_URL += '/'
+VALHALLA_HEADERS = {'Authorization': 'Token ' + str(VALHALLA_API_TOKEN)}
+VALIDATION_PROPOSAL = os.getenv('VALHALLA_VALIDATE_PROPOSAL', 'LCOSchedulerTest')
+
+
+def validate_userrequest(userrequests):
+    """
+        Call a valhalla instance validate endpoint with the userrequest dict. Raise an error with the error response
+        if it fails validation, otherwise return True
+    :param userrequest_dict:
+    :return: dictionary of validation errors
+    """
+    try:
+        ur_copy = copy.deepcopy(userrequests)
+        if isinstance(ur_copy, list):
+            for ur in ur_copy:
+                ur['proposal'] = VALIDATION_PROPOSAL
+        else:
+            ur_copy['proposal'] = VALIDATION_PROPOSAL
+        response = requests.post(VALHALLA_URL + 'validate/', json=ur_copy, headers=VALHALLA_HEADERS)
+        response.raise_for_status()
+        body = response.json()
+    except Exception as e:
+        body = {'request_durations': {}, 'errors': 'Problem validating with old observation portal: {}'.format(repr(e))}
+    return body
+
+
+def expand_cadence(userrequest):
+    """
+        Call a valhalla instance to expand a cadence userrequest into a set of requests, which can then be submitted
+        to the observation portal and converted into a requestgroup
+    :param userrequest:
+    :return: expanded userrequests
+    """
+    try:
+        ur_copy = copy.deepcopy(userrequest)
+        ur_copy['proposal'] = VALIDATION_PROPOSAL
+        response = requests.post(VALHALLA_URL + 'cadence/', json=ur_copy, headers=VALHALLA_HEADERS)
+        response.raise_for_status()
+        body = response.json()
+    except Exception as e:
+        body = {'errors': 'Problem validating with old observation portal: {}'.format(repr(e))}
+    return body
+
+
+def convert_userrequests_to_requestgroups(userrequests):
+    if isinstance(userrequests, list):
+        return [convert_userrequest_to_requestgroup(ur) for ur in userrequests]
+    else:
+        return convert_userrequest_to_requestgroup(userrequests)
+
+
+def convert_userrequest_to_requestgroup(userrequest):
+    requestgroup = userrequest
```

Comment:
It doesn't look like it matters in here, but any change made to `requestgroup` will also be made to `userrequest` since they are the same underlying object.